### PR TITLE
Correct pagination while displaying contacts

### PR DESCRIPTION
**Pull request by Shadow243**
Paging affects groups.

Ex: if in trusted Contacts there are 100 contacts and in Collected Contacts we have 2 contacts, the pagination will ensure that the 2 contacts in Collected are not displayed.

In this PR we group first then we paginate what is already grouped

### CODE REVIEW

**Comment by Shadow243** [comment]
There is an other issue reported by @Baraka24  and which concerns this PR.

File: modules/contacts/hm-contacts.php [hunk 1]
```diff
@@ -142,13 +142,66 @@ public function reset() {
...
+        $grouped = [];
+        foreach ($this->data as $value) {
+            
+            $column = null;
+            if (is_callable($func)) {
+                $column = call_user_func($func, $value);
+            } elseif (is_object($value)) {
+                $reflection = new ReflectionClass($value);
+                $property = $reflection->getProperty('data');
+                $property->setAccessible(true); // Make the private property accessible
```
Comment (kroky) [inline modules/contacts/hm-contacts.php:172]: There is a reason why private properties are private and using reflection to bypass this is something nasty. If this is a contact object isn't $value->value($_key) enough to get you what you want?

File: modules/contacts/hm-contacts.php [hunk 2]
```diff
@@ -142,13 +142,66 @@ public function reset() {
...
             return array();
         }
-        return array_slice($this->data, (($page - 1)*$size), $size, true);
+        if ($data === null) {
+            $data = $this->data;
+        }
+        return array_slice($data, (($page - 1)*$size), $size, true);
     }
 
+    public function group_by($column = 'group') {
```
Comment (kroky) [inline modules/contacts/hm-contacts.php:155]: Can you refactor the code to group by a string version of the column name? I don't see a reason we group by a callable or support cases for array or object contacts. You can refactor calling code to ensure we are working with a standard set of data - either array of arrays or array of contact objects.

**Comment by kroky** [comment]
Ah, yes, I didn't notice you pushed the updates. Looks good!

**Pull request merged**